static/js/components/dashboard/CourseDescription_test.js

```diff
@@ -31,80 +34,58 @@ export function findCourse(courseSelector: (course: Course, program: Program) =>
-      if (status === STATUS_ENROLLED_NOT_VERIFIED) {
-        assert.equal(text, "(enrolled)");
-      } else {
-        assert.equal(text, "");
-      }
-    }
-  });
-
-  it('shows failure text for failed courses', () => {
-    let failedCourse = findCourse(course => (
-      course.status === STATUS_NOT_OFFERED &&
-      course.runs[0].status === STATUS_NOT_PASSED
-    ));
-    const wrapper = shallow(<CourseDescription course={failedCourse} now={now} />);
-    assert.equal(wrapper.find(".course-description-result").text(), "You failed this course");
-  });
-
-  it('shows upgrade instructions for enrolled but not verified courses', () => {
-    let course = findCourse(course => course.status === STATUS_ENROLLED_NOT_VERIFIED);
-    const wrapper = shallow(<CourseDescription course={course} now={now} />);
-    assert.equal(
-      wrapper.find(".course-description-result").text(),
-      "You need to upgrade to the Verified course to get MicroMasters credit<IconButton />"
-    );
-  });
-
-  it('shows Complete for passed courses', () => {
-    let course = findCourse(course => course.status === STATUS_PASSED);
-    const wrapper = shallow(<CourseDescription course={course} now={now} />);
-    assert.equal(wrapper.find(".course-description-result").text(), "Complete!");
-  });
-
-  it('shows nothing for offered courses', () => {
-    let course = findCourse(course => course.status === STATUS_OFFERED_NOT_ENROLLED);
-    const wrapper = shallow(<CourseDescription course={course} now={now}/>);
-    assert.equal(wrapper.find(".course-description-result").text(), "");
-  });
-
-  it('shows Begins ... for verified courses when course start date is in future', () => {
-    let course = findCourse(course => course.status === STATUS_VERIFIED_NOT_COMPLETED);
-    let courseStart = moment(course.runs[0].course_start_date);
-    let yesterday = moment(courseStart).add(-1, 'days');
-    const wrapper = shallow(<CourseDescription course={course} now={yesterday}/>);
-    let formattedDate = courseStart.format(DASHBOARD_FORMAT);
-    assert.equal(wrapper.find(".course-description-result").text(), `Begins ${formattedDate}`);
-  });
-
-  it('shows nothing for verified courses when course start date is today or passed', () => {
-    let course = findCourse(course => course.status === STATUS_VERIFIED_NOT_COMPLETED);
-    let courseStart = moment(course.runs[0].course_start_date);
-    let tomorrow = moment(courseStart).add(1, 'days');
+  for (let status of statuses) {
+    it(`does show date with status ${status}`, () => {
+      let course = findCourse(course => course.status === status);
+      const wrapper = shallow(<CourseDescription course={course} />);
+      let firstRun: CourseRun = {};
 
-    // course starts today
-    let wrapper = shallow(<CourseDescription course={course} now={courseStart}/>);
-    assert.equal(wrapper.find(".course-description-result").text(), "");
+      if (course.runs.length > 0) {
```

Comment:
in unit tests, we should not be loading some test data, then doing something with that data if it happens to match some conditions. we should be using exact data that should result in some kind of behavior. that means the data should be created (or loaded) in its exact desired state as part of the test.

example: line 68 makes an assertion that an element has the text `Ended: ${formattedDate}`. the data that is intended to result in that label should be created or loaded exactly as part of the test. here's what i'm thinking in terms of code.

```javascript
export type CourseRunDateValues = {
  course_start_date?:     string;
  fuzzy_start_date?:     string;
  course_end_date?:     string;
};

let createCourseRun = (course: Course, status: string, dateValues: CourseRunDateValues) => {
  return Object.assign({
    "position": 1,
    "title": "title",
    "course_id": "course-v1:odl+GIO101+FALL14",
    "status": status,
    "id": 1
  }, dateValues);
};

it(`should show date with status ${status}`) {
  let course = _cloneDeep(findCourse(course => course.status === STATUS_PASSED));
  course.runs = [
    createCourseRun(course, STATUS_PASSED, {
      'course_end_date': '2016-09-09T10:20:10Z'
    })
  ];
  let firstRun = course.runs[0];
  const wrapper = shallow(<CourseDescription course={course} />);
  let courseEndDate = moment(firstRun.course_end_date);
  let formattedDate = courseEndDate.format(DASHBOARD_FORMAT);
  assert.equal(wrapper.find(".course-description-result").text(), `Ended: ${formattedDate}`);
}
```

we can load the course from `DASHBOARD_RESPONSE`, but we create our own course run as part of the test since the course run data is what influences the description text. i think your `switch` statement should be broken up into individual test cases (`it()` blocks) and data should be created explicitly like above